android/databases/src/main/java/com/coinbase/wallet/libraries/databases/db/Storage.kt

```diff
@@ -0,0 +1,249 @@
+package com.coinbase.wallet.libraries.databases.db
+
+import androidx.room.Room
+import androidx.sqlite.db.SimpleSQLiteQuery
+import com.coinbase.wallet.core.util.Optional
+import com.coinbase.wallet.libraries.databases.exceptions.DatabaseException
+import com.coinbase.wallet.libraries.databases.interfaces.DatabaseDaoInterface
+import com.coinbase.wallet.libraries.databases.interfaces.DatabaseModelObject
+import com.coinbase.wallet.libraries.databases.interfaces.StorageOptions
+import com.coinbase.wallet.libraries.databases.model.DatabaseOperation
+import com.coinbase.wallet.libraries.databases.model.DiskOptions
+import com.coinbase.wallet.libraries.databases.model.MemoryOptions
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.locks.Lock
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+
+/**
+ *  Room sqlitedb storage handler
+ *
+ *  @property modelDaos Mapping of database models to Dao. Exposed for inline functions below.
+ */
+class Storage<P : RoomDatabaseProvider> private constructor() {
+    private lateinit var options: StorageOptions
+    private lateinit var provider: RoomDatabaseProvider
+
+    /**
+     * Mapping of class to Observer. Exposed due to inline functions restriction
+     */
+    val observers = ConcurrentHashMap<Class<*>, PublishSubject<*>>()
```

Comment:
Can you make these `internal` and add `@PublishedApi` annotation to these (inline function magic I just learned)

Reply:
fixed